BUILD.md

```diff
@@ -0,0 +1,80 @@
+## Building CIL from source code
```

Comment:
Probably better to put this in the docs (contributing guide) instead?

Reply:
Do you mean:

1. I should rename the file?
2. The instructions should go in the documentation?

I normally like to see the build instructions in the repo

Reply:
I mean any one of:

- move to `docs/source/developer_guide.rst`
- leave in the original README section
- open a separate PR creating BUILD.md (not a fan)

Reply:
If this generates more trouble than it solves (reduce the length of the README), I will reinstate the info in the README and get rid of BUILD.md.

Reply:
reverted as discussed/agreed offline